ci-operator/step-registry/baremetalds/packet/setup/baremetalds-packet-setup-commands.sh

```diff
@@ -6,14 +6,81 @@ set -o pipefail
+    - packet_project_id: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-project-id') }}"
+    - packet_auth_token: "{{ lookup('file', lookup('env', 'CLUSTER_PROFILE_DIR') + '/packet-auth-token') }}"
+  tasks:
+  - name: write fix uid file
+    copy:
+      content: |
+        # Ensure our UID, which is randomly generated, is in /etc/passwd. This is required
+        # to be able to SSH.
+        if ! whoami &> /dev/null; then
+            if [ -x "\$(command -v nss_wrapper.pl)" ]; then
+                grep -v -e ^default -e ^\$(id -u) /etc/passwd > "/tmp/passwd"
+                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/tmp/passwd"
+                export LD_PRELOAD=libnss_wrapper.so
+                export NSS_WRAPPER_PASSWD=/tmp/passwd
+                export NSS_WRAPPER_GROUP=/etc/group
+            elif [[ -w /etc/passwd ]]; then
+                echo "\${USER_NAME:-default}:x:\$(id -u):0:\${USER_NAME:-default} user:\${HOME}:/sbin/nologin" >> "/etc/passwd"
+            else
+                echo "No nss wrapper, /etc/passwd is not writeable, and user matching this uid is not found."
+                exit 1
+            fi
+        fi
+      dest: "${SHARED_DIR}/fix-uid.sh"
+
+  - name: write Packet common configuration file
+    copy:
+      content: |
+        source "\${SHARED_DIR}/fix-uid.sh"
+
+        # Initial check
+        if [[ ! "\${CLUSTER_TYPE}" =~ ^packet.*$|^equinix.*$ ]]; then
+            echo >&2 "Unsupported cluster type '\${CLUSTER_TYPE}'"
+            exit 1
+        fi
+
+        IP=\$(cat "\${SHARED_DIR}/server-ip")
+        SSHOPTS=(-o 'ConnectTimeout=5' -o 'StrictHostKeyChecking=no' -o 'UserKnownHostsFile=/dev/null' -o 'ServerAliveInterval=90' -o LogLevel=ERROR -i "\${CLUSTER_PROFILE_DIR}/packet-ssh-key")
+
+        # Checkout packet server
+        for x in \$(seq 10) ; do
+            test "\${x}" -eq 10 && exit 1
+            ssh "\${SSHOPTS[@]}" "root@\${IP}" hostname && break
+            sleep 10
+        done
+      dest: "${SHARED_DIR}/packet-conf.sh"
+EOF
+ansible-playbook packet-setup.yaml -e "packet_hostname=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID}"  |& gawk '{ print strftime("%Y-%m-%d %H:%M:%S"), $0; fflush(); }'
+
+if [ $CLUSTER_TYPE == "equinix-ocp-metal" ] && [ -e "${CLUSTER_PROFILE_DIR}/ofcir_url" ] ; then
+    IPFILE=$SHARED_DIR/server-ip
+    CIRFILE=$SHARED_DIR/cir
+    if curl -kfX POST -H "Host: ofcir.apps.ostest.test.metalkube.org" $(cat ${CLUSTER_PROFILE_DIR}/ofcir_url)?name=ipi-${NAMESPACE}-${JOB_NAME_HASH}-${BUILD_ID} -o $CIRFILE ; then
+        jq -r .ip < $CIRFILE > $IPFILE
+        exit 0
+    fi
+fi
+
+
 # Avoid requesting a bunch of servers at the same time so they
 # don't race each other for available resources in a facility
 SLEEPTIME=$(( RANDOM % 120 ))
 echo "Sleeping for $SLEEPTIME seconds"
 sleep $SLEEPTIME
 
 # Run Ansible playbook
-cd
```

Comment:
The second part playbook is also named as the first one, which doesn't seem correct.
As an alternative approach, to avoid splitting the playbook, it could be possible to add the above code in a playbook task conditially executed (and also the same for the remaining ones)